server/container/util-java8/src/main/java/org/apache/james/util/SandglassExecutorService.java

```diff
@@ -0,0 +1,131 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.util;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+public class SandglassExecutorService implements ExecutorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SandglassExecutorService.class);
+    private static final int MAX_PARALLEL_THREADS = 10;
+
+    private final Duration sandglass;
+    private final ExecutorService executorService;
+
+    public SandglassExecutorService(Duration sandglass) {
```

Comment:
Why sandglass? Could you be more explicit?

Reply:
I will add documentation, the idea is that if you don't do the jobs during a delegated time you fail.

Reply:
I don't think that's an executor responsibility, but rather a task one. Like "TimeboundedTask" or something ?

Reply:
I don't understand why you did answered this question.
If you really want to keep your design, please do it with delegation so that the user can give you the executor it wants.